Update OpenMemory MCP configuration to build from Ushadow-io/mem0 for…

AnkushMalaker opened the pull request:
…k, add setup script for cloning, and enhance README with setup instructions.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Switched default memory backend to a new provider and added Neo4j-backed graph support.
  * UI endpoint moved to port 3333; additional service endpoints exposed for graph and admin.

* **Chores**
  * Deployment scripts now build images before startup and include health checks.
  * New environment variables for service credentials and connectors; added a service timeout setting.

* **Documentation**
  * Updated setup and troubleshooting steps with revised ports and startup workflow.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

AnkushMalaker committed:
Update OpenMemory MCP configuration to build from Ushadow-io/mem0 fork, add setup script for cloning, and enhance README with setup instructions.

AnkushMalaker committed:
Merge branch 'dev' into fix/open-memory-fork

thestumonkey commented on extras/openmemory-mcp/docker-compose.yml:
```diff
@@ -8,9 +8,11 @@ services:
   openmemory-mcp:
-    image: mem0/openmemory-mcp:latest
+    build:
+      context: ./mem0-fork/openmemory/api
```
where is this coming from? doens't match file system

AnkushMalaker replied:
https://github.com/chronicler-ai/chronicle/pull/228#issuecomment-3683853797 as explained here, its created during setup.sh flow

thestumonkey commented on extras/openmemory-mcp/setup.sh:
```diff
@@ -22,6 +22,26 @@ done
 echo "======================"
 
+# Clone the mem0 fork if not already present
+MEM0_DIR="mem0-fork"
```
Although this is fine for testing, I think that this is confusing and should just be in the openmemeory folder, as we already have many mem types

AnkushMalaker replied:
This is in the openmemory-mcp folder, I will rename the root level service as extras/openmemory
Is that what you're saying?

AnkushMalaker replied:
I'll rename in another PR as it'll show too many changes in this one

thestumonkey commented on extras/openmemory-mcp/docker-compose.yml:
```diff
@@ -8,9 +8,11 @@ services:
 
-  # OpenMemory MCP Server (official Docker image)
+  # OpenMemory MCP Server (built from Ushadow-io/mem0 fork)
   openmemory-mcp:
```
You need to use the compose file from the fork

thestumonkey reviewed (changes requested):
use the right compose values from openmemorry/docker compose
Should probably submodule in instead of cloning.

the folder naming is bad.  why is it openmemory-mcp?  it's not an mcp server.
"mem0-fork" is a terrible name.
can submodule it in or call it mem0-ushadow-fork

AnkushMalaker committed:
Update OpenMemory MCP configuration and setup

- Changed memory provider from `chronicle` to `openmemory_mcp` in `config.yml`.
- Updated server URL for OpenMemory MCP to use `http://host.docker.internal:8765`.
- Modified service path in `services.py` to reflect new directory structure.
- Enhanced `.env.template` with additional configuration options for OpenMemory.
- Removed obsolete `docker-compose.yml` file for OpenMemory MCP.
- Updated README with new service start instructions and port configurations.
- Improved setup script to configure the `.env` file correctly for OpenMemory.

AnkushMalaker committed:
Merge branch 'dev' into fix/open-memory-fork

AnkushMalaker committed:
Add Docker Compose setup for OpenMemory MCP and update configuration

- Introduced a new `docker-compose.yml` file to manage services for OpenMemory MCP, including Qdrant and Neo4j.
- Enhanced `.env.template` with additional configuration options for Neo4j and Qdrant.
- Updated `setup.sh` to reflect the use of pre-built Docker images and simplified the setup process.
- Improved user instructions for starting services and accessing various components.

AnkushMalaker committed:
Merge branch 'fix/open-memory-fork' of github.com:AnkushMalaker/friend-lite into fix/open-memory-fork
